test/python/circuit/test_singleton_gate.py

```diff
@@ -251,3 +253,26 @@ def test_positional_label(self):
         label_gate = SXGate("I am a little label")
         self.assertIsNot(gate, label_gate)
         self.assertEqual(label_gate.label, "I am a little label")
+
+    def test_immutable_pickle(self):
+        gate = SXGate()
+        self.assertFalse(gate.mutable)
+        with io.BytesIO() as fd:
+            pickle.dump(gate, fd)
+            fd.seek(0)
+            copied = pickle.load(fd)
+        self.assertFalse(copied.mutable)
```

Comment:
I think with this test, we potentially might want to assert a step further and say `gate is copied`?  If we're pickling within the same process that should hold - obviously not so in IPC, but `is` comparisons and the memory reduction aren't relevant over that.

Reply:
Updated in: https://github.com/Qiskit/qiskit/pull/10871/commits/c24edc1867b68a0c08c0ea5be3561e359387e3d3